modules/spotxBidAdapter.js

```diff
@@ -0,0 +1,330 @@
+    const hasVideoElement = videoSlotDiv.querySelectorAll('video').length;
+    const contentWidth = utils.getBidIdParameter('content_width', bid.params.video) ||
+      (hasVideoElement
+        ? videoSlotDiv.querySelectorAll('video')[0].offsetWidth
+        : videoSlotDiv.offsetWidth);
+    const contentHeight = utils.getBidIdParameter('content_height', bid.params.video) ||
+      (hasVideoElement
+        ? videoSlotDiv.querySelectorAll('video')[0].offsetHeight
+        : videoSlotDiv.offsetHeight);
+
+    const spotxImps = bidRequests.map(function(bid) {
+      const secure = isPageSecure || (utils.getBidIdParameter('secure', bid.params) ? 1 : 0);
+
+      const ext = {
+        player_width: contentWidth,
+        player_height: contentHeight,
+        sdk_name: 'Prebid 1+',
+        ad_mute: +!!utils.getBidIdParameter('ad_mute', bid.params.video),
+        hide_skin: +!!utils.getBidIdParameter('hide_skin', bid.params.video),
+        content_page_url: page,
+        versionOrtb: ORTB_VERSION,
+        bidId: bid.bidId,
+        videoSlot: bid.params.video.video_slot
+      };
+
+      if (utils.getBidIdParameter('ad_volume', bid.params.video) != '') {
+        ext.ad_volume = utils.getBidIdParameter('ad_volume', bid.params.video);
+      }
+
+      if (utils.getBidIdParameter('ad_unit', bid.params.video) != '') {
+        ext.ad_unit = utils.getBidIdParameter('ad_unit', bid.params.video);
+      }
+
+      if (utils.getBidIdParameter('outstreamFunction', bid.params.video) != '') {
+        ext.outstreamFunction = utils.getBidIdParameter('outstreamFunction', bid.params.video);
+      }
+
+      if (utils.getBidIdParameter('custom', bid.params.video) != '') {
+        ext.custom = utils.getBidIdParameter('custom', bid.params.video);
+      }
+
+      const mimes = utils.getBidIdParameter('mimes', bid.params.video) || ['application/javascript', 'video/mp4', 'video/webm'];
+
+      const spotxImp = {
+        id: Date.now(),
+        secure: secure,
+        video: {
+          w: contentWidth,
+          h: contentHeight,
+          ext: ext,
+          mimes: mimes
+        }
+      };
+
+      if (utils.getBidIdParameter('price_floor', bid.params) != '') {
+        spotxImp.bidfloor = utils.getBidIdParameter('price_floor', bid.params);
+      }
+
+      if (utils.getBidIdParameter('start_delay', bid.params.video) != '') {
+        spotxImp.video.startdelay = 0 + Boolean(utils.getBidIdParameter('start_delay', bid.params.video));
+      }
+
+      if (bid.crumbs && bid.crumbs.pubcid) {
+        pubcid = bid.crumbs.pubcid;
+      }
+
+      return spotxImp;
+    });
+
+    const language = navigator.language ? 'language' : 'userLanguage';
+    const device = {
+      h: screen.height,
+      w: screen.width,
+      dnt: utils.getDNT() ? 1 : 0,
+      language: navigator[language].split('-')[0],
+      make: navigator.vendor ? navigator.vendor : '',
+      ua: navigator.userAgent
+    };
+
+    let requestPayload = {
```

Comment:
This can be `const`, right? You don't change the reference anywhere

Reply:
We'll fix that in the next version. Thanks.